src/RedisResourceManager.php

```diff
@@ -279,6 +295,36 @@ protected function extractPassword($resource, $serverUri)
         return $server['pass'] ?? null;
     }
 
+    /**
+     * Extract password to be used on connection
+     *
+     * @param mixed $resource
+     * @param mixed $serverUri
+     * @return string|null
+     */
+    protected function extractUser($resource, $serverUri)
+    {
+        if (! empty($resource['user'])) {
+            return $resource['user'];
+        }
+
+        if (! is_string($serverUri)) {
+            return null;
+        }
+
+        // parse server from URI host{:?port}
+        $server = trim($serverUri);
+
+        if (strpos($server, '/') === 0) {
```

Comment:
you can use `str_starts_with` which is also more readable.
Just a nitpick, not a strong requirement tho.